src/Microsoft.Data/StringDataFrameColumn.cs

```diff
@@ -445,5 +445,24 @@ protected internal override Delegate GetDataViewGetter(DataViewRowCursor cursor)
 
         private ValueGetter<ReadOnlyMemory<char>> CreateValueGetterDelegate(DataViewRowCursor cursor) =>
             (ref ReadOnlyMemory<char> value) => value = this[cursor.Position].AsMemory();
+
+        protected internal override void AddValueUsingCursor(DataViewRowCursor cursor, DataViewSchema.Column schemaColumn)
+        {
+            long row = cursor.Position;
+            ReadOnlyMemory<char> value = default;
+            cursor.GetGetter<ReadOnlyMemory<char>>(schemaColumn)(ref value);
+            if (Length > row)
+            {
+                this[row] = new string(value.ToArray());
```

Comment:
Not sure if there's a way to avoid this copy

Reply:
call `value.ToString()`. 

[https://source.dot.net/#System.Private.CoreLib/shared/System/ReadOnlyMemory.cs,156](https://source.dot.net/#System.Private.CoreLib/shared/System/ReadOnlyMemory.cs,156)

[https://source.dot.net/#System.Private.CoreLib/shared/System/String.Manipulation.cs,1661](https://source.dot.net/#System.Private.CoreLib/shared/System/String.Manipulation.cs,1661)

If the value is the whole backing string, it will return the whole string without copying.

---
In reply to: [338807430](https://github.com/dotnet/corefxlab/pull/2766#discussion_r338807430) [](ancestors = 338807430)

Reply:
Interesting. Intellisense says this: `Returns the fully qualified type name of this instance` but I see that the method has different documentation on source.dot.net. I wonder why...